src/libraries/Microsoft.PowerFx.Core/Localization/ErrorResourceKey.cs

```diff
@@ -21,5 +22,24 @@ public ErrorResourceKey(string key)
 
             Key = key;
         }
+
+        public bool Equals(ErrorResourceKey other)
+        {
+            return other.Key == Key;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is ErrorResourceKey key && Equals(key);
+        }
+
+        public override int GetHashCode()
+        {
+            return Key.GetHashCode();
+        }
+
+        public static bool operator ==(ErrorResourceKey left, ErrorResourceKey right) => left.Equals(right);
```

Comment:
>left.Equals(right); [](http://example.com/codeflow?start=89&length=19)

Does this break (NRE) if left is null?

    ErrorResourceKey erk1 = MethodThatCanReturnNull();
    ErrorResourceKey erk2 = new ErrorResourceKey("sth");
    return erk1 == erk2;

Reply:
I corrected this even if this is not possible, per line 21